src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/entity/CaseEventMessageEntity.java

```diff
@@ -177,4 +181,27 @@ public Integer getRetryCount() {
     public void setRetryCount(Integer retryCount) {
         this.retryCount = retryCount;
     }
+
+
+    public static CaseEventMessageEntity buildMessageEntity(Map<String, Object> map, MessageState state) {
+        CaseEventMessageEntity entity = new CaseEventMessageEntity();
+
+        for (Map.Entry<String, Object> val : map.entrySet()) {
+            Object value = val.getValue();
+            switch (val.getKey()) {
+                case "messageId":
+                    entity.setMessageId((String) value);
+                    break;
+                case "eventTimeStamp":
+                    entity.setEventTimestamp(LocalDateTime.parse((String) value));
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        entity.setState(state);
+
+        return entity;
```

Comment:
Again more test code in the main prod branch?  Please remove.

Reply:
I've moved it here:  [CaseEventMessageEntityTest](https://github.com/hmcts/wa-case-event-handler/blob/13a6ce479c036f70c98187c224c0206a1cf91569/src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/entity/CaseEventMessageEntityTest.java)